tracer/src/Datadog.Trace.BenchmarkDotNet/DatadogExporter.cs

```diff
@@ -72,217 +61,203 @@ public void ExportToLog(Summary summary, ILogger logger)
-                        });
+                                testParameters.Metadata[TestTags.MetadataTestName] = benchmarkCase.DisplayInfo ?? string.Empty;
+                                test.SetParameters(testParameters);
+                            }
 
-                    if (report.BenchmarkCase.HasParameters)
-                    {
-                        var testParameters = new TestParameters
-                        {
-                            Arguments = new Dictionary<string, object>(),
-                            Metadata = new Dictionary<string, object>()
-                        };
-                        foreach (var parameter in report.BenchmarkCase.Parameters.Items)
-                        {
-                            var parameterValue = ClrProfiler.AutoInstrumentation.Testing.Common.GetParametersValueData(parameter.Value);
-                            if (testParameters.Arguments.TryGetValue(parameter.Name, out var currentValue))
+                            if (benchmarkTest.ResultStatistics is { } statistics)
                             {
-                                testParameters.Arguments[parameter.Name] += $"{currentValue}, {parameterValue}";
+                                double p90 = 0, p95 = 0, p99 = 0;
+                                if (statistics.Percentiles is { } percentiles)
+                                {
+                                    p90 = percentiles.P90;
+                                    p95 = percentiles.P95;
+                                    p99 = percentiles.Percentile(99);
+                                }
+
+                                test.AddBenchmarkData(
+                                    BenchmarkMeasureType.Duration,
+                                    "Duration of the benchmark",
+                                    new BenchmarkDiscreteStats(
+                                        statistics.N,
+                                        statistics.Max,
+                                        statistics.Min,
+                                        statistics.Mean,
+                                        statistics.Median,
+                                        statistics.StandardDeviation,
+                                        statistics.StandardError,
+                                        statistics.Kurtosis,
+                                        statistics.Skewness,
+                                        p99,
+                                        p95,
+                                        p90));
+
+                                test.SetTag("benchmark.runs", statistics.N);
                             }
-                            else
+
+                            if (benchmarkCase?.Config?.HasMemoryDiagnoser() == true)
                             {
-                                testParameters.Arguments[parameter.Name] = parameterValue;
+                                test.AddBenchmarkData(
```

Comment:
I can't follow what's changed here at all, github thinks it's all different. Just going to take your word for it, as the screenshot looks right 😉

Reply:
It's mostly the same code but now dates are calculated based on the `BenchmarkMetadata` object to fix the issue we are seeing in the flamegraph.